element-connector/src/test/java/org/eclipse/californium/elements/runner/LongTermRunner.java

```diff
@@ -0,0 +1,191 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations - initial implementation
+ ******************************************************************************/
+package org.eclipse.californium.elements.runner;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.eclipse.californium.elements.category.LongTerm;
+import org.junit.runner.notification.Failure;
+import org.junit.runner.notification.RunListener;
+import org.junit.runner.notification.RunNotifier;
+import org.junit.runners.BlockJUnit4ClassRunner;
+import org.junit.runners.model.InitializationError;
+
+/**
+ * Runner for tests under debugging.
+ * 
+ * Runs a test repeated until it fails or the maximum number of repeats is
+ * reached.
+ * 
+ * <pre>
+ * "org.eclipse.californium.elements.runner.LongTermRunner.repeats", maximum number of repeats for test.
+ * 0 := repeat test until failure. Default 100.
+ * "org.eclipse.californium.elements.runner.LongTermRunner.alife", interval in milliseconds.
+ * 0 := disabled. Default: 1000
+ * </pre>
+ * 
+ * @see LongTerm
+ */
+public class LongTermRunner extends BlockJUnit4ClassRunner {
```

Comment:
Should we chose the name according what it **does** or what it is **intended** to used for?
In both cases, the "other" need to be documented :-).